dask/dataframe/io/tests/test_sql.py

```diff
@@ -332,3 +335,98 @@ def test_no_character_index_without_divisions(db):
     # attempt to read the sql table with a character index and no divisions
     with pytest.raises(TypeError):
         read_sql_table("test", db, npartitions=2, index_col="name", divisions=None)
+
+
+def test_to_sql():
+    df_by_age = df.set_index("age")
+    df_appended = pd.concat([df, df,])
+
+    # Set this to False to use a sqlite DB file under the current directory, and not delete it (for easier debugging
+    # after tests have run)
+    rm_tmpdir = True
+    if rm_tmpdir:
+
+        @contextmanager
+        def tmp():
+            with tmpfile() as f:
+                yield "sqlite:///%s" % f
+
+    else:
+        from pathlib import Path
+
+        path = Path.cwd() / "tmp" / "sqlite"
+        path.parent.mkdir(exist_ok=True, parents=True)
+        uri = f"sqlite:///{path}"
+
+        @contextmanager
+        def tmp():
+            if path.exists():
+                path.unlink()
+            yield uri
+
+    for npartitions in [1, 2, 10]:
+        ddf = dd.from_pandas(df, npartitions)
+        ddf_by_age = ddf.set_index("age")
+
+        # Simple round trip test: use existing "number" index_col
+        with tmp() as uri:
+            ddf.to_sql("test", uri)
+            actual_ddf = read_sql_table("test", uri, "number").compute()
```

Comment:
You can actually test against the pandas read, which would have nullified the "do we need the index" question, above. I don't mind either.

Reply:
interesting idea. I kind of like testing that Dask round-trip explicitly, but can go either way